### ISSUE TITLE
Can't rename a property with a period

### ISSUE THREAD

**Issue by borov333** [orientechnologies/orientdb#6877]
## OrientDB Version, operating system, or hardware.
- v2.2.12

## Operating System
- [x] Linux
- [x] MacOSX

## Expected behavior and actual behavior
I'm trying to rename an existing class property, but it fails. See the terminal commands below:
```
orientdb {db=Customers}> info property SellCampaign.leadCategory               

PROPERTY 'SellCampaign.leadCategory'

Type.................: LINK
Mandatory............: true
Not null.............: true
Read only............: false
Default value........: null
Minimum value........: null
Maximum value........: null
REGEXP...............: null
Collate..............: {ODefaultCollate : name = default}
Linked class.........: SettingsLeadCategory
Linked type..........: null

orientdb {db=Customers}> ALTER PROPERTY SellCampaign.leadCategory NAME category
Error: com.orientechnologies.orient.core.exception.OCommandExecutionException: Property 'SellCampaign.leadCategory' not exists
	DB name="Customers"
	DB name="Customers"
```

It complaints that the property does not exist, but it does. The property is the embedded link to another class, but should it be a problem?

**Comment by borov333** [comment]
I just realized what the actual issue is. Please disregard my first statement on failing to rename a property. The issue exist, but it is for a different reason. During the renaming of the "leadCategory" property I accidentally renamed it to "SellCampaign.category". Yes, the full name with a period in the property name. So, even though the terminal was showing the property as existing it was only due to the fact that it cached the original schema, which is confusing sometimes. 

In reality the property was renamed, but now even worth, it has a period in the property name. At this point I can't do anything with this property. I can't rename it to anything and I can't even drop it. I tried in both, the terminal and the studio and get error each time. I tried to wrap the property with "`" characters, but to no avail.

Looks to me that once you set a property name with a period, you will get stuck on that name for ever. The only solution I found was to drop the class and recreate it from scratch, which is a pain if you have a lot of data with links and edges. I think either database shall not allow to name a property with a period at the first place, or allow to modify it's name if it has a period.

**Comment by luigidellaquila** [comment]
Hi @borov333 

I was able to reproduce the problem, I'll work on it asap

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @borov333 

I just pushed a fix for this on branch 2.2.x. It will be released with 2.2.13

Thanks

Luigi

**Comment by borov333** [comment]
Thanks!
